presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -244,16 +279,41 @@ private static boolean isPrefilled(HiveColumnHandle hiveColumnHandle)
                     HivePartitionKey partitionKey = partitionKeysByName.get(column.getName());
                     prefilledValue = getPrefilledColumnValue(column, partitionKey, path);
                 }
-                columnMappings.add(new ColumnMapping(column, prefilledValue, currentIndex));
+
+                boolean needCoercion = false;
+                HiveType coercionFrom = null;
+                if (columnCoercions.containsKey(column.getHiveColumnIndex())) {
+                    needCoercion = true;
+                    coercionFrom = columnCoercions.get(column.getHiveColumnIndex());
```

Comment:
when you switch to `Optional` this will become `coercionFrom = Optional.ofNullable(columnCoercions.get(column.getHiveColumnIndex()))`